### ISSUE TITLE
Can't create "circular referencing" Content Type MongoDB

### ISSUE THREAD

**Issue by pataiadam** [strapi/strapi#5385]
**Steps to reproduce the behavior**
1. Create a content type like this. This is not the concrete model I use, it is just an example: 

![image](https://user-images.githubusercontent.com/3729515/75695976-23529880-5cab-11ea-8b6a-9966c195c81d.png)

2. Create 2 categories (A and B)
3. Set B to A as subCategory (it works fine)
4. Now set A to B as subCategory (you can't set (it hangs), however after this action the categories explorer is empty 😮)

After this, in a few minutes my server has stopped with the following error:

```

<--- Last few GCs --->

[23682:0x1042a6000]   463716 ms: Scavenge 2037.8 (2052.0) -> 2036.6 (2052.0) MB, 13.1 / 0.0 ms  (average mu = 0.167, current mu = 0.007) allocation failure 
[23682:0x1042a6000]   463786 ms: Scavenge 2038.0 (2052.0) -> 2037.0 (2052.8) MB, 6.5 / 0.0 ms  (average mu = 0.167, current mu = 0.007) allocation failure 
[23682:0x1042a6000]   466512 ms: Mark-sweep 2038.5 (2052.8) -> 2037.4 (2053.0) MB, 2630.5 / 0.0 ms  (average mu = 0.136, current mu = 0.100) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x0737cfb80921 <JSObject>
    0: builtin exit frame: captureStackTrace(this=0x0737cfb81799 <JSFunction Error (sfi = 0x737748cc099)>,0x07379d371139 <MongooseTimeoutError map = 0x737422c5b71>,0x0737cfb81799 <JSFunction Error (sfi = 0x737748cc099)>)

    1: /* anonymous */(aka /* anonymous */) [0x7379d371001] [/Users/macbookpro/dev/strapi-corp/rep-comp-with-ref/node_modules/mongoose/lib/query.js:~4357] [pc=0xc01df10900e](th...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x100093bd5 node::Abort() [/usr/local/bin/node]
 2: 0x100093d9b void node::FPrintF<char const*&, char const*&>(__sFILE*, char const*, char const*&, char const*&) [/usr/local/bin/node]
 3: 0x1001d40f7 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0x1001d4093 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 5: 0x100356b65 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/usr/local/bin/node]
 6: 0x100358244 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/usr/local/bin/node]
 7: 0x100354fde v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/usr/local/bin/node]
 8: 0x100352fc6 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 9: 0x10035e94a v8::internal::Heap::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]
10: 0x10035e9d1 v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]
11: 0x10032bb85 v8::internal::Factory::AllocateRawArray(int, v8::internal::AllocationType) [/usr/local/bin/node]
12: 0x10032a3fd v8::internal::Factory::NewFixedArrayWithFiller(v8::internal::RootIndex, int, v8::internal::Object, v8::internal::AllocationType) [/usr/local/bin/node]
13: 0x10032d2fa v8::internal::Factory::NewFrameArray(int) [/usr/local/bin/node]
14: 0x100301937 v8::internal::(anonymous namespace)::CaptureStackTrace(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::(anonymous namespace)::CaptureStackTraceOptions) [/usr/local/bin/node]
15: 0x10030216e v8::internal::Isolate::CaptureAndSetSimpleStackTrace(v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::FrameSkipMode, v8::internal::Handle<v8::internal::Object>) [/usr/local/bin/node]
16: 0x100261c58 v8::internal::Builtin_Impl_ErrorCaptureStackTrace(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/usr/local/bin/node]
17: 0x1009a7eb9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [/usr/local/bin/node]
✨  Done in 471.81s.
```

**Expected behavior**
I would like to use this kind of "circular referencing" in my app and it should not cause any problems.

**System**
- Node.js version: 13.9.0
- NPM version: 6.14.1
- Strapi version:  v3.0.0-beta.18.7
- Database: mongo
- Operating system: osx

**Comment by pataiadam** [comment]
I found a temporary workaround:

Set  `"autoPopulate": false` to my sub-category components:

./components/cat/sub-category.json
```
...
  "attributes": {
    "category": {
      "autoPopulate": false,
      "model": "category"
    }
  },
...
```

But now (obviously) it is not populating the category :(

**Comment by lauriejim** [comment]
Hello @pataiadam !

Thank you for reporting this issue.

**Comment by MichelGeorgy** [comment]
I have the same issue. 
autoPopulate:false prevents the system from freezing, but the relation is not marked as selected in the select box.

**Comment by alexandrebodin** [comment]
Hi I'm going to close this issue as it was fixed in a recet release (v3.5.0) :)
